### ISSUE TITLE
MD024 `siblings_only: true` ignored in v0.32.1

### ISSUE THREAD

**Issue by Swiftb0y** [DavidAnson/markdownlint#1328]
Somewhere between v0.31.1 and v0.32.1 MD024 configured with `siblings_only: true` causes false positives. minimum reproducible example:
```yaml
# .markdownlint-cli2.yaml
config:
  default: true
  no-duplicate-header:
    siblings_only: true
```
```md
<!-- CHANGELOG.md -->
# Change log

## 1.0.0

### Features

## 2.0.0

### Features
```

resulting output (run using markdownlint-cli2):
```quote
markdownlint-cli2 v0.11.0 (markdownlint v0.32.1)
Finding: CHANGELOG.md
Linting: 1 file(s)
Summary: 1 error(s)
CHANGELOG.md:9 MD024/no-duplicate-heading Multiple headings with the same content [Context: "### Features"]
```

Expected result is for markdownlint-cli2 to just pass.
fyi, I'm running markdownlint-cli2 via pre-commit, but I don't think that should make a difference.

**Comment by DavidAnson** [comment]
The legacy "header" rule names were deprecated in version 32: https://github.com/DavidAnson/markdownlint/blob/main/CHANGELOG.md#0320

Changing "header" to "heading" in your example should produce the desired behavior.

**Comment by Swiftb0y** [comment]
Yup, indeed. sorry for missing that and wasting your time. It would be nice if there was a warning when detecting unused / deprecated rules. Its really easy to read past "header" and "heading" and not spot the difference, especially when at the end of a long rule name.

**Comment by DavidAnson** [comment]
This may help: https://github.com/DavidAnson/markdownlint/blob/main/schema/ValidatingConfiguration.md
